### DISCUSSION TITLE
Rate Limit

### DISCUSSION THREAD

**Discussion by AsifArmanRahman** [cinemagoer/cinemagoer#404] [Q&A]
Hi, I was wondering what's the rate limit of calls for the API?

**Comment by alberanid** [comment]
For a normal usage, to retrieve a few titles at a time for personal use, you should not hit any limit. However if you try to access many entries in a short time, you will surely be blocked. I do not know the exact rate limit.

Anyway, Cinemagoer is not thought to be a tool to mass-scan or mass-import the IMDb's data, and doing so would be against their terms of service. See https://raw.githubusercontent.com/cinemagoer/cinemagoer/master/DISCLAIMER.txt

If you need to access only some basic information that are also available from these data sets ( https://www.imdb.com/interfaces/ ) please use the "S3 data access system" of Cinemagoer: https://cinemagoer.readthedocs.io/en/latest/usage/s3.html

**Reply by AsifArmanRahman** [reply]
Is it **legal** to use the API in **production level** software's?

The project is personal, but I plan to host it eventually. And I require a bit more data available via the datasets, so I'll be using `http` as access system. For most case I need only the data returned using `main, release_dates`. However, note that I don't plan to mass-scan or anything.

**Reply by alberanid** [reply]
The usual disclaimer: I'm not a lawyer.

It is legal to use Cinemagoer to fetch the data and use it, as long as it's for personal and non-commercial purposes. This means that also (re)publishing the data is forbidden.

**Comment by alberanid** [comment]
This discussion is no longer relevant due to issue https://github.com/cinemagoer/cinemagoer/issues/557: as of June 2026 Cinemagoer no longer analyzes web pages and can only rely on non-commercial datasets.
